internal/runners/push/push.go

```diff
@@ -34,23 +42,51 @@ func NewPush(config *viper.Viper, prime primeable) *Push {
 	return &Push{config, prime.Output(), prime.Project()}
 }
 
-func (r *Push) Run() error {
+func (r *Push) Run(params PushParams) error {
 	if !authentication.Get().Authenticated() {
 		return locale.NewInputError("err_api_not_authenticated")
 	}
 
-	if r.project != nil && r.project.IsHeadless() {
+	if r.project == nil {
 		return errs.AddTips(locale.NewInputError(
 			"err_push_headless",
 			"You must first create a project."),
-			locale.Tl("push_headless_push_tip_conversion_url", "Visit [ACTIONABLE]{{.V0}}[/RESET] to create a project in the browser.", r.project.URL()),
-			locale.Tl("push_headless_push_tip_state_init", "Run [ACTIONABLE]state init[/RESET] to create a project with the State Tool.."),
+			locale.Tl("push_headless_push_tip_state_init", "Run [ACTIONABLE]state init[/RESET] to create a project with the State Tool."),
 		)
 	}
 
-	// Create the project remotely if it doesn't already exist
-	pjm, fail := model.FetchProjectByName(r.project.Owner(), r.project.Name())
+	owner := r.project.Owner()
+	name := r.project.Name()
+	if r.project.IsHeadless() {
+		namespace := params.Namespace
+		if !namespace.IsValid() {
+			names := projectfile.GetProjectNameForPath(r.config, filepath.Dir(r.project.Source().Path()))
+			if names == "" {
+				return errs.AddTips(
+					locale.NewInputError("push_needs_namespace", "Could not find out what project to push to."),
+					locale.Tl("push_add_namespace_tip", "You can specify a project by running [ACTIONABLE]state push <project>[/RESET]."),
+				)
+			}
+			var fail *failures.Failure
+			namespace, fail = project.ParseNamespace(names)
+			if fail != nil {
+				return errs.Wrap(fail.ToError(), "Could not parse namespace %s to push headless commit to", name)
+			}
+		}
+		owner = namespace.Owner
+		name = namespace.Project
+	} else {
+		if params.Namespace.IsValid() {
```

Comment:
```suggestion
		if !params.Namespace.IsValid() {
```

Reply:
Well, here we really have to check for `Namespace.IsValid()` to fulfill this AC: 

> If you provide a namespace while not in a headless state we should error out

But I noticed that I didn't specify the error message yet.  Ooops! Will fix.